module/Company/src/Company/Service/Company.php

```diff
@@ -199,6 +199,41 @@ public function getCategoryList($visible)
         return $this->getCategoryMapper()->findAll();
     }
 
+    public function getLabelList($visible)
+    {
+        $translator = $this->getTranslator();
+        if (!$visible) {
+            if (!$this->isAllowed('listAllLabels')) {
+                throw new \User\Permissions\NotAllowedException(
+                    $translator->translate('You are not allowed to access the admin interface')
+                );
+            }
+            $results = $this->getLabelMapper()->findAll();
+            return $this->getUniqueInArray($results, function ($a) {
+                return $a->getLanguageNeutralId();
+            });
+        }
+        if (!$this->isAllowed('listVisibleLabels')) {
+            throw new \User\Permissions\NotAllowedException(
+                $translator->translate('You are not allowed to list all labels')
+            );
+        }
+        if ($visible) {
```

Comment:
This is true here by definition. Since you always either throw an exception from the earlier `if (!$visible)` or return from it.

Reply:
Good point